provider/awssd/aws_sd_test.go

```diff
@@ -351,10 +351,31 @@ func TestAWSSDProvider_ApplyChanges(t *testing.T) {
 	endpoints, _ := provider.Records(ctx)
 	assert.True(t, testutils.SameEndpoints(expectedEndpoints, endpoints), "expected and actual endpoints don't match, expected=%v, actual=%v", expectedEndpoints, endpoints)
 
+	ctx = context.Background()
+
+	provider.ApplyChanges(ctx, &plan.Changes{
+		UpdateNew: []*endpoint.Endpoint{
+			{DNSName: "service1.private.com", Targets: endpoint.Targets{"1.2.3.4"}, RecordType: endpoint.RecordTypeA, RecordTTL: 60},
+		},
+		UpdateOld: []*endpoint.Endpoint{
+			{DNSName: "service1.private.com", Targets: endpoint.Targets{"1.2.3.4", "1.2.3.5"}, RecordType: endpoint.RecordTypeA, RecordTTL: 60},
+		},
+	})
```

Comment:
This doesn't add or remove `DNSName` values. It should verify that the code can add a new `DNSName` that didn't exist before, remove a `DNSName`, and add, remove, and delete the targets on an existing `DNSName`.

 It should also fail on the previous code. That means the stub `AWSSDClient` needs to assert that the `DNSName` for which the test is changing targets is not being deleted and targets in that `DNSName` that aren't being changed are  not being deleted.